src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java

```diff
@@ -11,168 +10,87 @@
+import org.fogbowcloud.saps.engine.core.archiver.storage.PermanentStorage;
+import org.fogbowcloud.saps.engine.core.archiver.storage.exceptions.PermanentStorageException;
+import org.fogbowcloud.saps.engine.core.archiver.storage.swift.SwiftPermanentStorage;
 import org.fogbowcloud.saps.engine.core.catalog.Catalog;
-import org.fogbowcloud.saps.engine.core.catalog.JDBCCatalog;
+import org.fogbowcloud.saps.engine.core.catalog.jdbc.JDBCCatalog;
 import org.fogbowcloud.saps.engine.core.model.SapsImage;
 import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
-import org.fogbowcloud.saps.engine.exceptions.SapsException;
 import org.fogbowcloud.saps.engine.utils.SapsPropertiesConstants;
+import org.fogbowcloud.saps.engine.utils.SapsPropertiesUtil;
 import org.fogbowcloud.saps.engine.utils.retry.CatalogUtils;
 
 public class Archiver {
 
     private final Properties properties;
     private final Catalog catalog;
-    private final SwiftAPIClient swiftAPIClient;
+    private final PermanentStorage permanentStorage;
     private final ScheduledExecutorService sapsExecutor;
-    private final boolean executionMode;
-
-    private static int MAX_ARCHIVE_TRIES = 1;
-    private static int MAX_SWIFT_UPLOAD_TRIES = 2;
+    private final boolean executionDebugMode;
 
     public static final Logger LOGGER = Logger.getLogger(Archiver.class);
 
-    public Archiver(Properties properties) throws SapsException, SQLException {
-        this(properties, new JDBCCatalog(properties), new SwiftAPIClient(properties), new ArchiverHelper());
-
-        LOGGER.info("Creating Archiver");
-        LOGGER.info("Imagestore " + properties.getProperty(SapsPropertiesConstants.IMAGE_DATASTORE_IP) + ":"
-                + properties.getProperty(SapsPropertiesConstants.IMAGE_DATASTORE_PORT));
+    public Archiver(Properties properties) throws ArchiverException, PermanentStorageException {
+        this(properties, new JDBCCatalog(properties));
     }
 
-    protected Archiver(Properties properties, Catalog catalog, SwiftAPIClient swiftAPIClient,
-                       ArchiverHelper archiverHelper) throws SapsException {
+    protected Archiver(Properties properties, Catalog catalog) throws ArchiverException, PermanentStorageException {
         if (!checkProperties(properties))
-            throw new SapsException("Error on validate the file. Missing properties for start Saps Controller.");
+            throw new ArchiverException("Error on validate the file. Missing properties for start Saps Controller.");
 
         this.properties = properties;
         this.catalog = catalog;
-        this.swiftAPIClient = swiftAPIClient;
         this.sapsExecutor = Executors.newScheduledThreadPool(1);
-        this.executionMode = properties.containsKey(SapsPropertiesConstants.SAPS_EXECUTION_DEBUG_MODE) && properties
+        this.permanentStorage = createStorageInstance(properties.getProperty(SapsPropertiesConstants.SAPS_PERMANENT_STORAGE_TYPE));
```

Comment:
As I told @wesleymonte. it seems better to receive the ref to the object.

Reply:
(i mean, a ref to the PermanentStorage)

Reply:
Done.